application.properties autocomplete is not working

With eclipse 4.27 and plugin version 4.27 on Ubuntu 22.04 when trying autocomplete application.properties firstly I am getting
```
An internal error occurred during: "MicroProfile properties collector".
org/jsoup/safety/Whitelist
```
Then :
```
An internal error occurred during: "MicroProfile properties collector".
Could not initialize class org.jboss.tools.quarkus.lsp4e.internal.ls.JavaDoc2MarkdownConverter
```
continuously
Here is log:
```
pr 13, 2023 6:05:26 PM org.eclipse.tm4e.core.registry.Registry _doLoadSingleGrammar
WARNING: No grammar source for scope [source.java-properties]
Apr 13, 2023 6:05:26 PM org.eclipse.tm4e.core.internal.rule.RuleFactory _compilePatterns
WARNING: CANNOT find grammar for scopeName [source.java-properties]. I am [source.quarkus-properties]
Apr 13, 2023 6:05:27 PM org.eclipse.lsp4mp.ls.MicroProfileLanguageServer initialize
INFO: Initializing MicroProfile server 0.6.0-20221201-1617 with /usr/lib/jvm/java-17-amazon-corretto
Apr 13, 2023 6:05:27 PM org.eclipse.lsp4mp.ls.properties.MicroProfileProjectInfoCache lambda$getProjectInfo$0
WARNING: Error while getting MicroProfileProjectInfo (classpath) for 'file:///home/sharofiddin/Projects/Quarkus/QuarkusKube/transaction-service/src/main/resources/application.properties'
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.commons.MicroProfileProjectInfo.getProjectURI()" because "delegate" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:212)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
	at org.eclipse.lsp4mp.ls.commons.ParentProcessWatcher.lambda$apply$0(ParentProcessWatcher.java:148)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.commons.MicroProfileProjectInfo.getProjectURI()" because "delegate" is null
	at org.eclipse.lsp4mp.extensions.ExtendedMicroProfileProjectInfo.<init>(ExtendedMicroProfileProjectInfo.java:78)
	at org.eclipse.lsp4mp.ls.properties.MicroProfileProjectInfoCache.lambda$getProjectInfoFromCache$1(MicroProfileProjectInfoCache.java:77)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	... 13 more

Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.core.PropertiesManager getMicroProfileProjectInfo
INFO: Start computing MicroProfile properties for '/home/sharofiddin/Projects/Quarkus/QuarkusKube/transaction-service' project.
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry loadExtensionProviders
INFO: ->- Loading .propertiesProviders extension point ->-
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: com.redhat.microprofile.jdt.internal.quarkus.core.properties.QuarkusCoreProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: com.redhat.microprofile.jdt.internal.quarkus.core.properties.QuarkusConfigRootProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: com.redhat.microprofile.jdt.internal.quarkus.core.properties.QuarkusConfigPropertiesProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: com.redhat.microprofile.jdt.internal.quarkus.core.properties.QuarkusConfigMappingProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: com.redhat.microprofile.jdt.internal.quarkus.scheduler.properties.QuarkusScheduledPropertiesProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: com.redhat.microprofile.jdt.internal.quarkus.kubernetes.properties.QuarkusKubernetesProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: com.redhat.microprofile.jdt.internal.quarkus.container.properties.QuarkusContainerImagesProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: com.redhat.microprofile.jdt.internal.quarkus.hibernate.properties.QuarkusHibernateORMProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: com.redhat.microprofile.jdt.internal.quarkus.cache.properties.QuarkusCacheResultProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: org.eclipse.lsp4mp.jdt.internal.jul.properties.JBossLogManagerPropertyProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: org.eclipse.lsp4mp.jdt.internal.config.properties.MicroProfileConfigPropertyProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: org.eclipse.lsp4mp.jdt.internal.config.properties.MicroProfileConfigPropertiesProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: org.eclipse.lsp4mp.jdt.internal.faulttolerance.properties.MicroProfileFaultToleranceProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: org.eclipse.lsp4mp.jdt.internal.reactivemessaging.properties.MicroProfileReactiveMessagingProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry addExtensionProviders
INFO:   Loaded propertiesProviders: org.eclipse.lsp4mp.jdt.internal.restclient.properties.MicroProfileRegisterRestClientProvider
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.AbstractMicroProfileProviderRegistry loadExtensionProviders
INFO: -<- Done loading .propertiesProviders extension point -<-
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.StaticPropertyProviderRegistry loadExtensionProviders
INFO: ->- Loading .staticPropertyProviders extension point ->-
Apr 13, 2023 6:05:28 PM org.eclipse.lsp4mp.jdt.internal.core.StaticPropertyProviderRegistry loadExtensionProviders
INFO: -<- Done loading .staticPropertyProviders extension point -<-
Apr 13, 2023 6:05:44 PM org.eclipse.lsp4mp.jdt.core.PropertiesManager getMicroProfileProjectInfo
INFO: End computing MicroProfile properties for '/home/sharofiddin/Projects/Quarkus/QuarkusKube/transaction-service' project in 16426ms.
```

> Your [quarkus-test.tar.gz](https://github.com/jbosstools/jbosstools-quarkus/files/11243274/quarkus-test.tar.gz) reproduce the error no?

Yes, it is reproducing the same error
logs:
```
Aug 04, 2023 3:10:54 PM org.eclipse.lsp4mp.jdt.core.PropertiesManager getMicroProfileProjectInfo
INFO: End computing MicroProfile properties for '/home/sharofiddin/Downloads/quarkus-test' project in 46966ms.
Aug 04, 2023 3:10:54 PM org.eclipse.lsp4mp.jdt.core.PropertiesManager getMicroProfileProjectInfo
INFO: End computing MicroProfile properties for '/home/sharofiddin/Downloads/quarkus-test' project in 10611ms.
Aug 04, 2023 3:10:54 PM org.eclipse.lsp4mp.ls.properties.MicroProfileProjectInfoCache lambda$getProjectInfo$0
WARNING: Error while getting MicroProfileProjectInfo (classpath) for 'file:///home/sharofiddin/Downloads/quarkus-test/src/main/resources/application.properties'
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.commons.MicroProfileProjectInfo.getProjectURI()" because "delegate" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:212)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
	at org.eclipse.lsp4mp.ls.commons.ParentProcessWatcher.lambda$apply$0(ParentProcessWatcher.java:148)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4mp.commons.MicroProfileProjectInfo.getProjectURI()" because "delegate" is null
	at org.eclipse.lsp4mp.extensions.ExtendedMicroProfileProjectInfo.<init>(ExtendedMicroProfileProjectInfo.java:78)
	at org.eclipse.lsp4mp.ls.properties.MicroProfileProjectInfoCache.lambda$getProjectInfoFromCache$1(MicroProfileProjectInfoCache.java:77)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
```

hi @Sharofiddin , taking back this issue just now.
i can reproduce the error by uncommenting the kubernetes dependency, and i have this stack : 
```
java.lang.NoClassDefFoundError: Could not initialize class org.jboss.tools.quarkus.lsp4e.internal.ls.JavaDoc2MarkdownConverter
	at org.jboss.tools.quarkus.lsp4e.internal.ls.JDTJavadocContentAccess.getMarkdownContentReader(JDTJavadocContentAccess.java:49)
	at org.jboss.tools.quarkus.lsp4e.internal.ls.JDTUtilsImpl.getJavadoc(JDTUtilsImpl.java:107)
	at com.redhat.microprofile.jdt.internal.quarkus.kubernetes.properties.QuarkusKubernetesProvider.collectProperties(QuarkusKubernetesProvider.java:153)
	at com.redhat.microprofile.jdt.internal.quarkus.kubernetes.properties.QuarkusKubernetesProvider.processClass(QuarkusKubernetesProvider.java:110)
	at org.eclipse.lsp4mp.jdt.core.AbstractTypeDeclarationPropertiesProvider.collectProperties(AbstractTypeDeclarationPropertiesProvider.java:58)
	at org.eclipse.lsp4mp.jdt.core.PropertiesManager.collectProperties(PropertiesManager.java:261)
	at org.eclipse.lsp4mp.jdt.core.PropertiesManager$1.acceptSearchMatch(PropertiesManager.java:237)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.report(MatchLocator.java:2196)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.reportMatching(MatchLocator.java:3158)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.reportMatching(MatchLocator.java:2877)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.process(MatchLocator.java:2048)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1364)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1401)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1520)
	at org.eclipse.jdt.internal.core.search.JavaSearchParticipant.locateMatches(JavaSearchParticipant.java:135)
	at org.eclipse.jdt.internal.core.search.BasicSearchEngine.findMatches(BasicSearchEngine.java:251)
	at org.eclipse.jdt.internal.core.search.BasicSearchEngine.search(BasicSearchEngine.java:602)
	at org.eclipse.jdt.core.search.SearchEngine.search(SearchEngine.java:670)
	at org.eclipse.lsp4mp.jdt.core.PropertiesManager.scanJavaClasses(PropertiesManager.java:221)
	at org.eclipse.lsp4mp.jdt.core.PropertiesManager.getMicroProfileProjectInfo(PropertiesManager.java:145)
	at org.eclipse.lsp4mp.jdt.core.PropertiesManager.getMicroProfileProjectInfo(PropertiesManager.java:113)
	at org.eclipse.lsp4mp.jdt.core.PropertiesManager.getMicroProfileProjectInfo(PropertiesManager.java:104)
	at org.jboss.tools.quarkus.lsp4e.QuarkusLanguageClient.lambda$2(QuarkusLanguageClient.java:81)
	at org.eclipse.core.runtime.jobs.Job$2.run(Job.java:187)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoClassDefFoundError: org/jsoup/safety/Whitelist [in thread "Worker-14: MicroProfile properties collector"]
	at com.overzealous.remark.Remark.<init>(Remark.java:83)
	at org.jboss.tools.quarkus.lsp4e.internal.ls.JavaDoc2MarkdownConverter.<clinit>(JavaDoc2MarkdownConverter.java:42)
	... 25 more
```
looking at it.

--- a/.github/workflows/build_test.yml
+++ b/.github/workflows/build_test.yml
@@ -37,6 +37,7 @@ jobs:
         architecture: x64
         distribution: 'temurin'
         cache: maven
+
     # Install Maven 3.9
     - name: Set up Maven
       uses: stCarolas/setup-maven@07fbbe97d97ef44336b7382563d66743297e442f #v4.5
@@ -57,6 +58,7 @@ jobs:
       uses: GabrielBB/xvfb-action@86d97bde4a65fe9b290c0b3fb92c2c4ed0e5302d #v1
       with:
         run: >
+          export MAVEN_OPTS="-Xms1G -Xmx2G"
           mvn clean verify -U -fae --settings build-ci/maven-settings.xml -DskipITests=true
           -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true -ntp
 

--- a/.github/workflows/build_test.yml
+++ b/.github/workflows/build_test.yml
@@ -55,12 +55,7 @@ jobs:
 
     # Build and compile using Maven
     - name: Build/Compile and run unit tests
-      uses: GabrielBB/xvfb-action@86d97bde4a65fe9b290c0b3fb92c2c4ed0e5302d #v1
-      with:
-        run: >
-          export MAVEN_OPTS="-Xms1G -Xmx2G"
-          mvn clean verify -U -fae --settings build-ci/maven-settings.xml -DskipITests=true
-          -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true -ntp
+      run: xvfb-run -a export MAVEN_OPTS="-Xms1G -Xmx2G"; mvn clean verify -U -fae -B --settings build-ci/maven-settings.xml -DskipITests=true -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true -ntp
 
     # Archive artifacts to be applied in Publish Reports workflow
     - name: Archiving test artifacts

--- a/.github/workflows/build_test.yml
+++ b/.github/workflows/build_test.yml
@@ -55,7 +55,7 @@ jobs:
 
     # Build and compile using Maven
     - name: Build/Compile and run unit tests
-      run: xvfb-run -a export MAVEN_OPTS="-Xms1G -Xmx2G"; mvn clean verify -U -fae -B --settings build-ci/maven-settings.xml -DskipITests=true -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true -ntp
+      run: xvfb-run -a mvn clean verify -U -fae -B --settings build-ci/maven-settings.xml -DskipITests=true -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true -ntp
 
     # Archive artifacts to be applied in Publish Reports workflow
     - name: Archiving test artifacts

--- a/plugins/org.jboss.tools.quarkus.lsp4e/META-INF/MANIFEST.MF
+++ b/plugins/org.jboss.tools.quarkus.lsp4e/META-INF/MANIFEST.MF
@@ -15,7 +15,6 @@ Require-Bundle: org.eclipse.lsp4e,
  org.eclipse.jdt.core.manipulation,
  org.eclipse.jdt.ui,
  org.eclipse.lsp4mp.jdt.core,
- org.jsoup;bundle-version="1.8.3";visibility:=reexport,
  org.eclipse.ui,
  org.jboss.tools.foundation.ui,
  org.apache.commons.lang3,
@@ -35,6 +34,7 @@ Bundle-ClassPath: .,
  server/org.eclipse.lsp4mp.ls-uber.jar,
  server/com.redhat.quarkus.ls.jar,
  lib/remark.jar,
+ lib/jsoup.jar,
  server/com.redhat.qute.ls-uber.jar
 Bundle-Activator: org.jboss.tools.quarkus.lsp4e.QuarkusLSPPlugin
 Export-Package: org.jboss.tools.quarkus.lsp4e

--- a/plugins/org.jboss.tools.quarkus.lsp4e/pom.xml
+++ b/plugins/org.jboss.tools.quarkus.lsp4e/pom.xml
@@ -81,6 +81,11 @@
                                         <artifactId>remark</artifactId>
                                         <version>1.2.0</version>
                                     </artifactItem>
+                                    <artifactItem>
+                                        <groupId>org.jsoup</groupId>
+                                        <artifactId>jsoup</artifactId>
+                                        <version>1.9.2</version>
+                                    </artifactItem>
                                 </artifactItems>
                             </configuration>
                         </execution>

--- a/plugins/org.jboss.tools.quarkus.lsp4e/src/org/jboss/tools/quarkus/lsp4e/internal/ls/HtmlToPlainText.java
+++ b/plugins/org.jboss.tools.quarkus.lsp4e/src/org/jboss/tools/quarkus/lsp4e/internal/ls/HtmlToPlainText.java
@@ -23,7 +23,7 @@
  */
 package org.jboss.tools.quarkus.lsp4e.internal.ls;
 
-import org.jsoup.internal.StringUtil;
+import org.jsoup.helper.StringUtil;
 import org.jsoup.nodes.Element;
 import org.jsoup.nodes.Node;
 import org.jsoup.nodes.TextNode;
@@ -46,7 +46,9 @@ public class HtmlToPlainText {
 	 */
 	public String getPlainText(Element element) {
 		FormattingVisitor formatter = new FormattingVisitor();
-		NodeTraversor.traverse(formatter, element); // walk the DOM, and call .head() and .tail() for each node
+		NodeTraversor traversor = new NodeTraversor(formatter);
+		traversor.traverse(element); // walk the DOM, and call .head() and .tail() for each node
+
 		return formatter.toString();
 	}
 

--- a/plugins/org.jboss.tools.quarkus.lsp4e/src/org/jboss/tools/quarkus/lsp4e/internal/ls/JavaDoc2MarkdownConverter.java
+++ b/plugins/org.jboss.tools.quarkus.lsp4e/src/org/jboss/tools/quarkus/lsp4e/internal/ls/JavaDoc2MarkdownConverter.java
@@ -17,7 +17,7 @@ import java.lang.reflect.Field;
 
 import org.jboss.tools.quarkus.lsp4e.QuarkusLSPPlugin;
 import org.jsoup.safety.Cleaner;
-import org.jsoup.safety.Safelist;
+import org.jsoup.safety.Whitelist;
 
 import com.overzealous.remark.Options;
 import com.overzealous.remark.Options.Tables;
@@ -50,7 +50,7 @@ public class JavaDoc2MarkdownConverter extends AbstractJavaDocConverter {
 			Field safelistField = Cleaner.class.getDeclaredField("safelist");
 			safelistField.setAccessible(true);
 
-			Safelist s = (Safelist) safelistField.get(c);
+			Whitelist s = (Whitelist) safelistField.get(c);
 
 			s.addProtocols("a", "href", "file", "jdt");
 			s.addProtocols("img", "src", "file");

